stdlib/2and3/csv.pyi

```diff
@@ -18,19 +20,8 @@ from _csv import (
 from collections import OrderedDict
 from typing import Any, Dict, Iterable, Iterator, List, Mapping, Optional, Sequence, Text, Type, Union
 
-_Dialect = Union[str, Dialect, Type[Dialect]]
 _DictRow = Mapping[str, Any]
 
-class Dialect(object):
```

Comment:
I thought this was actually an unrelated class at runtime? If so, we should probably keep it here, but accept both the _csv and the csv Dialects below.

Reply:
They are separate classes at runtime, but are perhaps morally one class, since all the CPython code duck types around it.
The issue is `_csv.writer` also needs to accept `csv.Dialect` so that common code like `csv.writer(f, dialect=csv.excel)` is accepted.
I guess we could get around the cyclic import with `if TYPE_CHECKING`? Does that work in pyi / is that better?

Reply:
Cyclic imports are supported in stubs. if TYPE_CHECKING is not necessary as it's implied.

Reply:
We discussed this some more; it seems best to just treat both as one class because that's how they're used in practice.